blog/tidy-finance-dummy-data/index.qmd

```diff
@@ -0,0 +1,422 @@
+
+```{r}
+beta_dummy <- stock_panel_monthly |> 
+  mutate(
+    beta_monthly = rnorm(n(), mean = 1, sd = 1),
+    beta_daily = beta_monthly + rnorm(n()) / 100
+  )
+
+dbWriteTable(
+  tidy_finance_dummy,
+  "beta", 
+  beta_dummy, 
+  overwrite = TRUE
+)
+```
+
+### Dummy `compustat` table
+
+To create dummy firm characteristics, we take all columns from the `compustat` table and create random numbers between 0 and 1 using the same trick as with the macro data tables. For simplicity, we set the `datadate` for each firm-year observation to the last day of the year, although it is empirically not the case. 
+
+```{r}
+relevant_columns <- tbl(tidy_finance, "compustat") |> 
+  select(-c(gvkey, datadate, year)) |> 
+  names()
+
+commands <- unlist(
+  map(
+    relevant_columns, 
+    ~rlang::exprs(!!..1 := runif(n()))
+  )
+)
+
+compustat_dummy <- stock_panel_yearly |> 
+  mutate(
+    datadate = ymd(str_c(year, "12", "31")),
+    !!!commands
+  )
+
+dbWriteTable(
+  tidy_finance_dummy, 
+  "compustat", 
+  compustat_dummy,
+  overwrite = TRUE
+)
+```
+
+### Dummy `crsp_monthly` table
+
+The `crsp_monthly` table only lacks a few more columns compared to `stock_panel_monthly`: the returns `ret` drawn from a normal distribution, the excess returns `ret_excess` with small deviations from the returns, the shares outstanding `shrout` and the last price per month `altprc` both drawn from uniform distributions, and the market capitalization `mktcap` as the product of `shrout` and `altprc`.
+
+```{r}
+tbl(tidy_finance, "crsp_monthly")
+
+crsp_monthly_dummy <- stock_panel_monthly |> 
+  mutate(
+    date = ceiling_date(month, "month") - 1,
+    ret = pmax(rnorm(n()), -1),
+    ret_excess = pmax(ret - runif(n(), 0, 0.0025), -1),
+    shrout = runif(n(), 1, 50) * 1000,
+    altprc = runif(n(), 0, 1000),
+    mktcap = shrout * altprc
+  ) |> 
+  group_by(permno) |> 
+  arrange(month) |> 
+  mutate(mktcap_lag = lag(mktcap)) |> 
+  ungroup()
+
+dbWriteTable(
+  tidy_finance_dummy, 
+  "crsp_monthly",
+  crsp_monthly_dummy,
+  overwrite = TRUE
+)
+```
+
+### Dummy `crsp_daily` table
+
+The `crsp_daily` table only contains a `month` column and the daily excess returns `ret_excess` as additional columns to `stock_panel_daily`.  
+
+```{r}
+crsp_daily_dummy <- stock_panel_daily |> 
+  mutate(
+    month = floor_date(date, "month"),
+    ret_excess = pmax(rnorm(n()), -1)
+  )
+
+dbWriteTable(
+  tidy_finance_dummy,
+  "crsp_daily",
+  crsp_daily_dummy, 
+  overwrite = TRUE
+)
+```
+
+## Create bond dummy data
+
+Lastly, we move to the bond data that we use in our books. 
+
+### Dummy `mergent` data
+
+To create dummy data with the structure of Mergent FISD, we calculate the empirical probabilities of actual bonds for several variables: `maturity`, `offering_amt`, `interest_frequency`, `coupon`, and `sic_code`. We use these prbabilities to sample a small cross section of bonds with completely made up `complete_cusip`, `issue_id` and `issuer_id`. 
```

Comment:
Typo correction: "We use these **probabilities** to sample a small **cross-section** of bonds with completely made up `complete_cusip`, `issue_id`**,** and `issuer_id`."

Reply:
One beautiful Oxford comma is missing - in case you do not see it.